### PR TITLE
Respect website architecture

### LINKED ISSUE #77: Disk: Stop flat archive creation
hi :)

loving this program so far but just had one feature idea / request, is it possible to have an like a -mirror flag or sitemap output flag so the downloader can output with the same folder structure as the site based on url or folders that its downloading.  the reason i ask is when trying this on a few public open directories it makes several folders with the site name and then with the file name and creates folders for each file / page. 

here is an example

https://releases.ubuntu.com/19.10/
https://releases.ubuntu.com/20.04/

so we could setup suckit to output to releases.ubuntu.com/19.10/, releases.ubuntu.com/20.04 and  etc 

Thanks

### PR DESCRIPTION
**Pull request by Skallwar**
Closes #77

### CODE REVIEW

File: src/scraper.rs [hunk 1]
```diff
@@ -82,15 +84,14 @@ impl Scraper {
         }
     }
 
-    /// Fix the URLs contained in the DOM-tree so they point to each other
-    fn fix_domtree(&self, old_url_str: &mut String, new_url: &Url) {
-        let path_map = self.path_map.lock().unwrap();
-        let path = path_map.get(new_url.as_str()).unwrap();
+    /// Fix the URLs contained in the DOM-tree so they point to each other relatif to each other
```
Comment (CohenArthur) [inline src/scraper.rs:87]:
```suggestion
    /// Fix the URLs contained in the DOM-tree so they point to each other relatively
```

**Review by CohenArthur — APPROVED** [review]

File: tests/fixtures/mod.rs [hunk 2]
```diff
@@ -62,41 +64,58 @@ fn check_auth_credentials(auth_header: Option<&Header>) -> bool {
...
+    //
+    //         let head_stdout = head_output_child.wait_with_output().unwrap();
+    //         sort_output_child.wait().unwrap();
+    //         return Ok(String::from_utf8(head_stdout.stdout)
+    //             .unwrap()
+    //             .trim()
+    //             .parse()
+    //             .unwrap());
+    //     }
+    // }
```
Comment (CohenArthur) [inline tests/fixtures/mod.rs:118]: Dead code?

File: tests/fixtures/mod.rs [hunk 3]
```diff
@@ -62,41 +64,58 @@ fn check_auth_credentials(auth_header: Option<&Header>) -> bool {
...
-            let head_output_child = Command::new("wc")
-                .args(&["-w"])
-                .stdin(sort_output)
-                .stdout(Stdio::piped())
-                .spawn()
-                .unwrap();
+    let cmd = {
+        Exec::shell(format!("ls {}", dir))
+            | Exec::shell(format!("grep '{}'", pattern))
+            | Exec::shell("wc -l")
```
Comment (CohenArthur) [inline tests/fixtures/mod.rs:72]: Maybe add a comment to indicate that it's an actual shell pipe

File: tests/fixtures/mod.rs (hunk 2 above)
Comment (Skallwar) [reply tests/fixtures/mod.rs:118]: Yes :sweat_smile:

File: tests/fixtures/mod.rs (hunk 3 above)
Comment (Skallwar) [reply tests/fixtures/mod.rs:72]: Ok

File: tests/fixtures/mod.rs [hunk 4]
```diff
@@ -67,6 +65,7 @@ pub fn get_file_count_with_pattern(pattern: &str, dir: &str) -> Result<usize, ()
     // Command being run: `ls | grep pattern | wc -w`
 
     let cmd = {
+        // Pipe ('|') is overloaded here and do a real pipe
```
Comment (CohenArthur) [inline tests/fixtures/mod.rs:68]:
```suggestion
        // Pipe ('|') is overloaded here and does a real pipe
```

**Review by CohenArthur — APPROVED** [review]
Perfect! All good for me if you just fix the tiny grammar typo in the pipe comment! Really looking forward to seeing this merged !

**Pull request merged**